feat(rdb): seamless upgrade v (#2435)
scaleway/docs-content: managed-databases/postgresql-and-mysql/api-cli/migrating-endpoints.mdx

---
meta:
  title: Migrating Database Instance endpoints via the Scaleway CLI
  description: This page explains how to migrate Database Instance endpoints via the Scaleway CLI.
content:
  h1: Migrating Database Instance endpoints via the Scaleway CLI
  paragraph: This page explains how to migrate Database Instance endpoints via the Scaleway CLI.
tags: managed-database postgresql mysql engine version database engine endpoints
dates:
  validation: 2023-11-22
  posted: 2023-11-22
categories:
  - managed-databases
  - postgresql-and-mysql
---

> You can migrate your endpoints from a Database Instance to another anytime using the Scaleway CLI anytime.

Reviewer: ```suggestion
You can migrate your endpoints from a Database Instance to another anytime, using the Scaleway CLI.
```

The following commands are useful when you [upgrade the version of your database engine](/managed-databases/postgresql-and-mysql/how-to/upgrade-version) via the **Upgrade only** option. In this case your endpoint must be migrated manually.

<Macro id="iam-requirements" />

<Message type="requirement">
  - You have an account and are logged into the [Scaleway console](https://console.scaleway.com)
  - You have a [PostgreSQL or MySQL Database Instance](/managed-databases/postgresql-and-mysql/how-to/create-a-database)
  - You have installed and set up the [Scaleway Command Line Interface (CLI)](/developer-tools/scaleway-cli/quickstart)
</Message>

1. Retrieve a list of your Database Instances.
    ```
    scw rdb instance list
    ```
    You should get a response like the following:

    ```
    ID                                    NAME             NODE TYPE
    70ca91f9-d649-4485-9107-e73c1ecf1eab  rdb-calm-outland  db-play2-pico
    0317a95d-924e-40c0-b6a3-ed579ef3e157  rdb-calm-inland  db-play2-pico
    ```

>     Take note of the UUID of the Database Instance whose endpoint you want to migrate, and of the Database Instance you want to migrate it to.

Reviewer 2: ```suggestion
    Make a record of the unique identifier (UUID) of the Database Instance that you wish to migrate from, as well as the UUID of the target Database Instance you intend to migrate to.
```

---
meta:
  title: Migrating Database Instance endpoints via the Scaleway CLI
  description: This page explains how to migrate Database Instance endpoints via the Scaleway CLI.
content:
  h1: Migrating Database Instance endpoints via the Scaleway CLI
  paragraph: This page explains how to migrate Database Instance endpoints via the Scaleway CLI.
tags: managed-database postgresql mysql engine version database engine endpoints
dates:
  validation: 2023-11-22
  posted: 2023-11-22
categories:
  - managed-databases
  - postgresql-and-mysql
---

You can migrate your endpoints from a Database Instance to another anytime, using the Scaleway CLI.

The following commands are useful when you [upgrade the version of your database engine](/managed-databases/postgresql-and-mysql/how-to/upgrade-version) via the **Upgrade only** option. In this case your endpoint must be migrated manually.

<Macro id="iam-requirements" />

<Message type="requirement">
  - You have an account and are logged into the [Scaleway console](https://console.scaleway.com)
  - You have a [PostgreSQL or MySQL Database Instance](/managed-databases/postgresql-and-mysql/how-to/create-a-database)

>   - You have installed and set up the [Scaleway Command Line Interface (CLI)](/developer-tools/scaleway-cli/quickstart)

Reviewer 3: - You must have jq library installed on your machine. 

- The destination database must not have an already configured endpoint of the same type. 
  - if you want to migrate a `load_balancer` endpoint type the destination instance must not have a `load_balancer` endpoint. You can use the CLI to remove the already existing endpoint on the destination instance 
  - if you want to migrate a `private_network` endpoint type the destination instance must not have a `private_network` endpoint. You can use the CLI to remove the already existing endpoint on the destination instance

</Message>

1. Retrieve a list of your Database Instances.
    ```
    scw rdb instance list
    ```
    You should get a response like the following:

    ```
    ID                                    NAME             NODE TYPE
    70ca91f9-d649-4485-9107-e73c1ecf1eab  rdb-calm-outland  db-play2-pico
    0317a95d-924e-40c0-b6a3-ed579ef3e157  rdb-calm-inland  db-play2-pico
    ```

    Make a record of the unique identifier (UUID) of the Database Instance that you wish to migrate from, as well as the UUID of the target Database Instance you intend to migrate to.
2. Run the following command to retrieve the endpoint of your Database Instance. Make sure you replace `$DB_INSTANCE_ID` with the UUID of the original Database Instance.
    ```

>     pub_endpoint=$(scw rdb instance get $DB_INSTANCE_ID -o json | jq -r ".endpoints[] | select( .load_balancer != null ) | .id")

Reviewer 3: this command will migrate only the load_balancer endpoint i.e the public access. 

It is also possible to select a `private_network` endpoint id by using the following command 

```
priv_endpoint=$(scw rdb instance get $DB_INSTANCE_ID -o json | jq -r ".endpoints[] | select( .private_network != null ) | .id")
```

    ```
    ID                                    NAME             NODE TYPE
    70ca91f9-d649-4485-9107-e73c1ecf1eab  rdb-calm-outland  db-play2-pico
    0317a95d-924e-40c0-b6a3-ed579ef3e157  rdb-calm-inland  db-play2-pico
    ```

    Make a record of the unique identifier (UUID) of the Database Instance that you wish to migrate from, as well as the UUID of the target Database Instance you intend to migrate to.
2. Run the following command to retrieve the public endpoint of your Database Instance. Make sure you replace `$DB_INSTANCE_ID` with the UUID of the original Database Instance.
    <Message type="important">
      Before migrating the endpoint, keep in mind that:
      - If you want to migrate a `load_balancer` endpoint type the target Database Instance must not have a `load_balancer` endpoint
      - If you want to migrate a `private_network` endpoint type the target Database Instance must not have a `private_network` endpoint

      You can use the CLI to [remove the already existing endpoints](https://github.com/scaleway/scaleway-cli/blob/master/docs/commands/rdb.md#delete-a-database-instance-endpoint) on the target Database Instance.
    </Message>

    ```

>     pub_endpoint=$(scw rdb instance get $DB_INSTANCE_ID -o json | jq -r ".endpoints[] | select( .load_balancer != null ) | .id")

Reviewer 3: may be use the endpoint variable here and bellow so the migrate command will work for bith cases

```suggestion
    endpoint=$(scw rdb instance get $DB_INSTANCE_ID -o json | jq -r ".endpoints[] | select( .load_balancer != null ) | .id")
```

    ```

    <Message type="tip">
      To retrieve a Private Network endpoint, run:
      ```

>       priv_endpoint=$(scw rdb instance get $DB_INSTANCE_ID -o json | jq -r ".endpoints[] | select( .private_network != null ) | .id")

Reviewer 3: ```suggestion
      endpoint=$(scw rdb instance get $DB_INSTANCE_ID -o json | jq -r ".endpoints[] | select( .private_network != null ) | .id")
```

      ```
      Make sure you replace `$DB_INSTANCE_ID` with the UUID of the original Database Instance.
    </Message>

3. Run the following command to migrate the endpoint.
    ```

>     scw rdb endpoint migrate endpoint-id=$pub_endpoint instance-id=$NEW_INSTANCE_ID

Reviewer 3: ```suggestion
    scw rdb endpoint migrate endpoint-id=$endpoint instance-id=$NEW_INSTANCE_ID
```
